test/hive-connect.hcl

```diff
@@ -0,0 +1,345 @@
+        command = "hiveserver"
+      }
+
+      resources {
+        cpu    = 500
+        memory = 1024
+      }
+
+      logs {
+        max_files     = 10
+        max_file_size = 2
+      }
+
+      template {
+        data = <<EOH
+          HIVE_SITE_CONF_hive_metastore_uris="thrift://{{ env "NOMAD_UPSTREAM_ADDR_hive-metastore" }}"
+          HIVE_SITE_CONF_hive_execution_engine="mr"
+          HIVE_SITE_CONF_hive_support_concurrency=false
+          HIVE_SITE_CONF_hive_driver_parallel_compilation=true
+          HIVE_SITE_CONF_hive_metastore_warehouse_dir="s3a://hive/warehouse"
+          HIVE_SITE_CONF_hive_metastore_event_db_notification_api_auth=false
+          HIVE_SITE_CONF_hive_server2_active_passive_ha_enable=true
+          HIVE_SITE_CONF_hive_server2_enable_doAs=false
+          HIVE_SITE_CONF_hive_server2_thrift_port=10000
+          HIVE_SITE_CONF_hive_server2_thrift_bind_host="127.0.0.1"
+          HIVE_SITE_CONF_hive_server2_authentication="NOSASL"
+          CORE_CONF_fs_defaultFS = "s3a://default"
+          CORE_CONF_fs_s3a_connection_ssl_enabled = false
+          CORE_CONF_fs_s3a_endpoint = "http://{{ env "NOMAD_UPSTREAM_ADDR_minio" }}"
+          CORE_CONF_fs_s3a_path_style_access = true
+          EOH
+
+        destination = "local/config.env"
+        env         = true
+      }
+
+      template {
+        data = <<EOH
+          CORE_CONF_fs_s3a_access_key = "minioadmin"
+          CORE_CONF_fs_s3a_secret_key = "minioadmin"
+          EOH
+
+        destination = "secrets/.env"
+        env         = true
+      }
+    }
+  }
+
+  group "metastore" {
+    count = 1
+
+    service {
+      name = "hive-metastore"
+      port = 9083
+
+      check {
+        name     = "beeline"
+        type     = "script"
+        task     = "metastoreserver"
+        command  = "/bin/bash"
+        args     = ["-c", "beeline -u jdbc:hive2:// -e \"SHOW DATABASES;\" &> /tmp/check_script_beeline_metastoreserver && echo \"return code $?\""]
+        interval = "30s"
+        timeout  = "120s"
+      }
+
+      connect {
+        sidecar_service {
+          proxy {
+            upstreams {
+              destination_name = "hive-database"
+              local_bind_port  = 5432
+            }
+            upstreams {
+              destination_name = "minio"
+              local_bind_port  = 9000
+            }
+          }
+        }
+      }
+    }
+
+    network {
+      mode = "bridge"
+    }
+
+    task "waitfor-hive-database" {
+      lifecycle {
+        hook    = "prestart"
+      }
+
+      driver = "docker"
+      config {
+        image = "alioygur/wait-for:latest"
+        args = [ "-it", "${NOMAD_UPSTREAM_ADDR_hive-database}", "-t", 120 ]
+      }
+    }
+    task "waitfor-minio" {
+      lifecycle {
+        hook    = "prestart"
+      }
+
+      driver = "docker"
+      config {
+        image = "alioygur/wait-for:latest"
+        args = [ "-it", "${NOMAD_UPSTREAM_ADDR_minio}", "-t", 120 ]
+      }
+    }
```

Comment:
We should change this to something that actually checks minio, no?

Reply:
Yeah, I have an idea to build REST service which query consul by (e.g.) service name and verify that all health checks for target service are green, then exit:0 . But it is also a separate issue.

Reply:
@zhenik I'll have a quick stab at it.